### PR TITLE
Update scalar dl manual deployment guide for auditor service on EKS 

### PR DESCRIPTION
**Pull request by scalar-boney**
Update the scalar dl manual deployment guide for auditor deployment. The auditor service is an optional service.

Configuration file related changes are available in https://github.com/scalar-labs/scalar-kubernetes/pull/150

### CODE REVIEW

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+* You should confirm that the replica count of the Auditor and Envoy pods in the `scalardl-audit-custom-values.yaml` file is equal to the number of nodes in the scalardlpool.
+* You should keep an equal number of pods for Envoy, Ledger and Auditor.
+
+### Steps
+
+1. Download the following Scalar DL configuration files from the [scalar-kubernetes](https://github.com/scalar-labs/scalar-kubernetes/tree/audit-guide-aks/conf) repository.
+Note that they are going to be versioned in the future, so you might want to change the branch to use a proper version.
+    * Scalardl-audit-custom-values.yaml
+    * Schema-loading-custom-values.yaml
```
Comment (ymorimo) [inline docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:79]:
```suggestion
    * scalardl-audit-custom-values.yaml
    * schema-loading-custom-values.yaml
```

File: docs/ManualDeploymentGuideScalarDLOnAWS.md [hunk 2]
```diff
@@ -222,12 +246,21 @@ When you need to remove the resources that you have created, remove the resource
...
+   helm delete load-schema
 
-    # Uninstall Scalar DL with a release name 'my-release-scalardl'
-    helm delete my-release-scalardl
+   # Uninstall Scalar DL with a release name 'my-release-scalardl'
+   helm delete my-release-scalardl
    ```
+You can uninstall Scalar Auditor with the following Helm commands:
+
+    ```console
```
Comment (ymorimo) [inline docs/ManualDeploymentGuideScalarDLOnAWS.md:257]: This code block is not properly formatted for the same reason as https://github.com/scalar-labs/scalar-kubernetes/pull/150/files#r743349918

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 1 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Thank you for your suggestions. 
Document formatted.

File: docs/ManualDeploymentGuideScalarDLOnAWS.md (hunk 2 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLOnAWS.md]: Thank you for your suggestions. 
Document formatted.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+### Steps
+
+* Follow step 1 to step 3 in the [Deploy Scalar DL on AWS](./ManualDeploymentGuideScalarDLOnAWS.md) guide to create a Scalar DL Auditor environment.
+
+## Step 2. Peer the Virtual Networks
+
+This section shows how to peer the VPC for Scalar DL deployment.
+
+In this guide, Ledger and Auditor are deployed on the private subnet of the different VPCs, 
+so you need to add peering for internal communication between the Auditor, Ledger and Client SDK(application).
```
Comment (feeblefakie) [inline docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:47]: Logically weird.

The current logic is as follows but isn't it weird? Why client suddenly appears?
Ledger and Auditor are deployed in different VPCs => peering between Auditor, Ledger, and Client.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+* You must create 3 peering between 3 Virtual Networks.
+
+### Recommendations
+* You should create Network ACLs for Ledger and Auditor VPCs.
+* You should restrict all access from the Auditor and Client except scalardl envoy LoadBalancer ports (50051 and 50052) in the Ledger Network ACLs.
+* You should restrict all access from the Ledger and Client except scalardl auditor envoy LoadBalancer ports (40051 and 40052) in the Auditor Network ACLs.
+
+### Steps
+
+* Create the peering between the Ledger and Auditor VPCs on the basis of [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/create-vpc-peering-connection.html) with the above requirements.
```
Comment (feeblefakie) [inline docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:60]: Aren't they redundant?

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+* Create the peering between the Ledger and Client VPCs on the basis of [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/create-vpc-peering-connection.html) with the above requirements.
+* Create the peering between the Auditor and Client VPCs on the basis of [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/create-vpc-peering-connection.html) with the above requirements.
+* Update Ledger route tables with Auditor and Client VPC peering connections based on [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/vpc-peering-routing.html).
+* Update Auditor route tables with Ledger and Client VPC peering connections based on [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/vpc-peering-routing.html).
+* Update Client route tables with Ledger and Auditor VPC peering connections based on [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/vpc-peering-routing.html).
+* Add new Inbound and Outbound rule to restrict unwanted access to Ledger and Auditor.
```
Comment (feeblefakie) [inline docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:66]: These are very unorganized and redundant.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+* [Helm](https://helm.sh/docs/intro/install/): Helm command-line tool to manage releases in the EKS cluster, Helm version 3.2.1 or latest is required. In this guide, it is used to deploy Scalar DL Auditor and Schema loading helm charts to the EKS cluster.
+* You must create a Github Personal Access Token (PAT) on the basis of Github official document with `read:packages` scope, it is used to access the `scalar-auditor` and `scalardl-schema-loader` container images from GitHub Packages.
+
+### Requirements
+
+* You must have the authority to pull `scalar-auditor` and `scalardl-schema-loader` container images.
+
+### Recommendations
+
+* You should confirm that the replica count of the Auditor and Envoy pods in the `scalardl-audit-custom-values.yaml` file is equal to the number of nodes in the `scalardlpool`.
```
Comment (feeblefakie) [inline docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:101]: These two are saying the same thing?
Also, `you should confirm`  sounds very weird. If we should set, then say so. Only confirming is recommended?

Moreover, we should talk about the reason for the recommendation.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+* Create the peering between the Ledger and Client VPCs on the basis of [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/create-vpc-peering-connection.html) with the above requirements.
+* Create the peering between the Auditor and Client VPCs on the basis of [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/create-vpc-peering-connection.html) with the above requirements.
+* Update Ledger route tables with Auditor and Client VPC peering connections based on [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/vpc-peering-routing.html).
+* Update Auditor route tables with Ledger and Client VPC peering connections based on [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/vpc-peering-routing.html).
+* Update Client route tables with Ledger and Auditor VPC peering connections based on [AWS official guide](https://docs.aws.amazon.com/vpc/latest/peering/vpc-peering-routing.html).
+* Add new Inbound and Outbound rule to restrict unwanted access to Ledger and Auditor.
+    * Add new Inbound rules to the Ledger Network ACLs to allow the Ledger Envoy LoadBalancer (50051, 50052) access from the Auditor and Client.
```
Comment (feeblefakie) [inline docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:67]: 50051 and 50052 are only default values and might be changed.

It should be something like this.
```the ports the Ledger listens to (e.g., 50051 and 50052 by default)```

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+In this guide, Ledger and Auditor are deployed on the private subnet of the different VPCs, 
+so you need to add peering for internal communication between the Auditor, Ledger and Client SDK(application).  
+
+### Requirements
+
+* You must create 3 peering between 3 Virtual Networks.
+
+### Recommendations
+* You should create Network ACLs for Ledger and Auditor VPCs.
+* You should restrict all access from the Auditor and Client except scalardl envoy LoadBalancer ports (50051 and 50052) in the Ledger Network ACLs.
```
Comment (feeblefakie) [inline docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:55]: The sentence is not understandable as English and is unclear.
It's talking about the ACL of Ledger?
As a sentence, it should probably be something like this. 

* You should set up the Network ACLs of the Ledger to restrict unused accesses.

Also, 50051 and 50052 are only default values so we should say so.

**Review by feeblefakie — CHANGES_REQUESTED** [review]

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 5 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:66]: I'm sorry, 
 
 Before, Hiro san said `If we need 3 peerings, have 3 steps here. `
  https://github.com/scalar-labs/scalar-kubernetes/pull/150#discussion_r745191191

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 4 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:60]: Before, Hiro san said If we need 3 peerings, have 3 steps here.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 5 above)
Comment (feeblefakie) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:66]: I did not say `have redundant 3 steps here`

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 4 above)
Comment (feeblefakie) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:60]: I did not say `have redundant 3 steps here`.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
+# Deploy Scalar DL Auditor on AWS
+
+This guide shows you how to manually deploy Scalar DL Auditor on a managed database service and a managed Kubernetes service in AWS as part of deploying Scalar DL for production.
+If you have not deployed Scalar DL Ledger, please follow [Deploy Scalar DL on AWS](./ManualDeploymentGuideScalarDLOnAWS.md) guide.
+
+## What we create
+
+![image](images/eks-auditor-arch.png)
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:8]: I have created this diagram based on https://github.com/scalar-labs/scalar-kubernetes/blob/audit-guide-eks/docs/images/network_diagram_eks.png .

Red and blue colors are additionally added.
Red color for indicating auditor environment and deep violet color used for indicating peering.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+* A Bastion instance with a public IP
+* Amazon CloudWatch
+
+## Prerequisites
+
+* Scalar DL Ledger deployment with Auditor configuration completed.
+
+## Step 1. Create an environment
+
+Scalar DL Auditor component is implemented to detect Byzantine faults,
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Thank you for your suggestion.
The above suggestion is integrated.

[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: > Isn't it too late to explain Auditor here?

Yes

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+it manages the identical states of Ledger so you need to deploy Ledger and Auditor in separate administrative domains. For more details, please refer to [Getting Started with Scalar DL Auditor](https://github.com/scalar-labs/scalardl/blob/master/docs/getting-started-auditor.md) guide.
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Thank you for your suggestion.
The above suggestion is integrated.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 12]
```diff
@@ -0,0 +1,174 @@
+# Deploy Scalar DL Auditor on AWS
+
+This guide shows you how to manually deploy Scalar DL Auditor on a managed database service and a managed Kubernetes service in AWS as part of deploying Scalar DL for production.
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:4]: Thank you for your suggestion.
The above suggestion is integrated.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+## Prerequisites
+
+* Scalar DL Ledger deployment with Auditor configuration completed.
+
+## Step 1. Create an environment
+
+Scalar DL Auditor component is implemented to detect Byzantine faults,
+it manages the identical states of Ledger so you need to deploy Ledger and Auditor in separate administrative domains. For more details, please refer to [Getting Started with Scalar DL Auditor](https://github.com/scalar-labs/scalardl/blob/master/docs/getting-started-auditor.md) guide.
+
+In this guide, for ease of explanation, we deploy Auditor in a different cluster on the same subscription. However, it is highly recommended to deploy it in another administrative domain in production.
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Thank you for your suggestion.
The above suggestion is integrated.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+
+### Requirements
+
+* You must create VPCs with different CIDR ranges.
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: > Different from what?

Different from Ledger and Client.

> Are we creating multiple VPCs? (It says VPCs not a VPC)

No, Only one VPC for Auditor.

Document updated with the following requirements
```
You must create a VPC for Auditor with a different IP address range that is not used by Ledger and Client.
```

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+
+* You should deploy Ledger and Auditor in separate administrative domains.
+
+### Steps
+
+* Follow step 1 to step 3 in the [Deploy Scalar DL on AWS](./ManualDeploymentGuideScalarDLOnAWS.md) guide to create a Scalar DL Auditor environment.
+
+## Step 2. Peer the Virtual Networks
+
+This section shows how to peer the VPC for Scalar DL deployment.
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: > Which is the VPC?
To peering, we need two VPCs but it says the VPC. It's not understandable.

Ledger, AUditor and Client.
Yes, 
Sorry, It should be VPCs

 
This section is no longer available.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+
+In this guide, Ledger and Auditor are deployed on the private subnet of the different VPCs,
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Sorry for the inconvenience.
Redundant sentence removed.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 3 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Sorry for the inconvenience.

> The current logic is as follows but isn't it weird?

Yes, It is weird.

> Why client suddenly appears?

I have updated the above sentence more clearly.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 4 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Thank you for your reply.
Removed redundant sentences.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 5 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:64]: Thank you for your reply.
Removed redundant sentences.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 6 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: > These two are saying the same thing?

No, the First one is for avoiding resource shortage.
Second one for better performance (If we use 3 Ledger recommended to use 3 auditors)

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+
+* Follow step 1 to step 3 in the [Deploy Scalar DL on AWS](./ManualDeploymentGuideScalarDLOnAWS.md) guide to create a Scalar DL Auditor environment.
+
+## Step 2. Peer the Virtual Networks
+
+In this guide, Ledger, Auditor, and Client applications are deployed on the private subnet of the different VPCs, 
+so you need to add peering for internal communication between the Auditor, Ledger, and Client application.  
+
+### Requirements
+* You must create peering between the Ledger and Auditor VPCs.
```
Comment (feeblefakie) [inline docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md:48]: As you mentioned, these are not requirements. Requirements should be states.

**Review by feeblefakie — CHANGES_REQUESTED** [review]

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
+* A managed database service
+    * DynamoDB
+* A Bastion instance with a public IP
+* Amazon CloudWatch
+
+## Prerequisites
+
+* Scalar DL Ledger deployment with Auditor configuration completed.
+
+## Step 1. Create an environment
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Document updated with environment details.
Requirements were also added.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 7 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Thank you for your suggestion.
Document updated with the above change.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 8 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Thank you for your suggestion.
Document updated with the above suggestion.

File: docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md (hunk 17 above)
Comment (scalar-boney) [reply docs/ManualDeploymentGuideScalarDLAuditorOnAWS.md]: Thank you for your comments.
Document updated with new requirements.

**Pull request merged**